iothub/device/src/Exceptions/ExceptionHandlingHelper.cs

```diff
@@ -11,7 +11,21 @@ namespace Microsoft.Azure.Devices.Client
 {
     internal class ExceptionHandlingHelper
     {
-        public static IDictionary<HttpStatusCode, Func<HttpResponseMessage, Task<Exception>>> GetDefaultErrorMapping()
+        /// <summary>
+        /// Mapper to map IoT hub service returned 6 digit error codes to <see cref="IotHubClientException.ErrorCode"/>.
+        /// </summary>
+        /// <param name="errorCode">The 6 digit error code returned by IoT hub service.</param>
+        /// <returns>The IotHubClientErrorCode that is returned as a part of IotHubClientException.</returns>
+        internal static IotHubClientErrorCode GetIotHubClientErrorCode(string errorCode)
+        {
+            return errorCode switch
+            {
+                "400004" => IotHubClientErrorCode.ArgumentInvalid,
```

Comment:
I have just this one error code mapped for now. We can add on to this as we encounter other cases.